### DISCUSSION TITLE
new option in "BlueRetro Presets config"

### DISCUSSION THREAD

**Discussion by davidxgames** [darthcloud/BlueRetro#338] [Q&A]
hello, could you please add to "BlueRetro Presets config" can you put an option for a control that I made from n64 based on BLE GamePad? I'm going to share it, it would serve to recycle controls.

         ["PAD_LD_LEFT", "C-Left",
         ["PAD_LD_RIGHT", "C-Right",
         ["PAD_LD_DOWN", "C-Down",
         ["PAD_LD_UP", "C-Up",
         ["PAD_LX_LEFT", "Stick left",
         ["PAD_LX_RIGHT", "Stickright",
         ["PAD_LY_DOWN", "Stick down",
         ["PAD_LY_UP", "Stick up",
         ["PAD_MQ", "D-pad left",
         ["PAD_RT", "D-pad right",
         ["PAD_RB_RIGTH", "D-pad down",
         ["PAD_LT", "D-pad up",
         ["PAD_RB_LEFT", "B",
         ["PAD_RB_DOWN", "A",
         ["PAD_MM", "Start",
         ["PAD_RB_UP", "Z",
         ["PAD_LS", "L",
         ["PAD_RS", "R",

**Comment by darthcloud** [comment]
Is that for publicly shared implementation of BLE GamePad?

If not, anyone making it's own will endup with different mapping needs. So the value to have it in the preset would be low.

**Reply by davidxgames** [reply]
yes i will share the firmware and schematics

**Reply by darthcloud** [reply]
If you are making the FW a better solution is to make your controller work with the default preset.

**Reply by davidxgames** [reply]
I've already made one for snes, and didn't have any problems, but some n64 buttons are hard to match with the default settings. But if you can't, I'll have to add it to my tutorial on how to map the buttons, thanks!

**Reply by darthcloud** [reply]
I means I can but it's much better if you handle it though your FW and not force people to use a preset.

You need to simply create a fake 2nd joystick with usage Rx and Ry map the C buttons to it. Use the hat switch for the D-pad.

@GamingNJncos can probably help you if needed

**Reply by davidxgames** [reply]
The problem is that I don't know how to program! hahaha. I have achieved it based on self-learning, but I will keep trying. Mapping the buttons the control works very well. I will ask for help to solve it without having to map, thanks!!!

**Reply by darthcloud** [reply]
Make you code public, then it's easy for people to help with it.

**Reply by GamingNJncos** [reply]
@davidxgames Agreed, if you can share your code I can take a look.

**Reply by davidxgames** [reply]
@GamingNJncos  How do I contact you?
